contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.1.0--3.2.0.sql

```diff
@@ -721,6 +721,158 @@ CREATE OR REPLACE VIEW sys.spt_tablecollations_view AS
     WHERE
         c.is_sparse = 0 AND p.attnum >= 0;
 
+--- Dropping function sys.datetime2fromparts since there is a change in return type
+DROP FUNCTION IF EXISTS sys.datetime2fromparts(NUMERIC, NUMERIC, NUMERIC, NUMERIC, NUMERIC, NUMERIC, NUMERIC, NUMERIC) CASCADE;
```

Comment:
do we need drop if we are doing create or replace? also if a view is referencing this function, drop could fail due to dependency. It also indicated that dependency testing is missing for this function in upgrade tests

Reply:
Removing this drop statement in upgrade script causing error:

Upgrade Complete
----------------
Optimizer statistics are not transferred by pg_upgrade.
Once you start the new server, consider running:
    /home/runner/psql_target/bin/vacuumdb -U runner --all --analyze-in-stages

Running this script will delete the old cluster's data files:
    ./delete_old_cluster.sh
pg_upgrade completed!
Updating babelfish extensions...
waiting for server to start.... done
server started
ERROR:  cannot change return type of existing function
ALTER EXTENSION
HINT:  Use DROP FUNCTION datetime2fromparts(numeric,numeric,numeric,numeric,numeric,numeric,numeric,numeric) first.
Error: Process completed with exit code 1.

Reply:
Yeah, because it's drop cascade , and the replace function is replace the function itself only

so we have to figure out what will cascadely be dropped together with this drop in upgrade script

Reply:
However, adding drop statement is causing any view calling sys.datetime2fromparts() to throw "Function not found" error.

Reply:
we have standard procedure of doing rename following create and replace. Please look for rename pattern in upgrade file and follow that.

Reply:
Done